experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/lazyField.spec.ts

```diff
@@ -65,3 +92,314 @@ describe("lazyField", () => {
+				cursor,
+				rootFieldAnchor,
+			);
+
+			assert.equal(field.length, 0);
+		});
+
+		it("With value", () => {
+			const builder = new SchemaBuilder("test");
+			const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
+			const rootSchema = SchemaBuilder.fieldOptional(numberLeafSchema);
+
+			const { context, cursor } = createSingleValueTree(builder, rootSchema, 42);
+
+			const field = new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(numberLeafSchema),
+				cursor,
+				rootFieldAnchor,
+			);
+
+			assert.equal(field.length, 1);
+		});
+	});
+
+	/**
+	 * Creates a single-node, primitive tree, and returns a field associated with that node.
+	 */
+	function createPrimitiveField(
+		kind: ValueSchema,
+		initialTree?:
+			| SchemaAware.TypedField<FieldSchema, SchemaAware.ApiMode.Flexible>
+			| readonly ITreeCursorSynchronous[]
+			| ITreeCursorSynchronous,
+	): LazyOptionalField<[TreeSchema<"leaf">]> {
+		const builder = new SchemaBuilder("test");
+		const leafSchema = builder.leaf("leaf", kind);
+		const rootSchema = SchemaBuilder.fieldOptional(leafSchema);
+
+		const { context, cursor } = createSingleValueTree(builder, rootSchema, initialTree);
+
+		return new LazyOptionalField(
+			context,
+			SchemaBuilder.fieldOptional(leafSchema),
+			cursor,
+			rootFieldAnchor,
+		);
+	}
+
+	function createStructField(
+		initialTree?:
+			| SchemaAware.TypedField<FieldSchema, SchemaAware.ApiMode.Flexible>
+			| readonly ITreeCursorSynchronous[]
+			| ITreeCursorSynchronous,
+	): LazyOptionalField<[TreeSchema<"struct">]> {
+		const builder = new SchemaBuilder("test");
+		const booleanLeafSchema = builder.leaf("bool", ValueSchema.Boolean);
+		const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
+		const leafSchema = builder.struct("struct", {
+			foo: SchemaBuilder.fieldValue(booleanLeafSchema),
+			bar: SchemaBuilder.fieldOptional(numberLeafSchema),
+		});
+		const rootSchema = SchemaBuilder.fieldOptional(leafSchema);
+
+		const { context, cursor } = createSingleValueTree(builder, rootSchema, initialTree);
+
+		return new LazyOptionalField(
+			context,
+			SchemaBuilder.fieldOptional(leafSchema),
+			cursor,
+			rootFieldAnchor,
+		);
+	}
+
+	describe("map", () => {
+		it("boolean", () => {
+			const field = createPrimitiveField(ValueSchema.Boolean, false);
+
+			assert.deepEqual(
+				field.map((value) => value),
+				[false],
+			);
+		});
+
+		it("number", () => {
+			const field = createPrimitiveField(ValueSchema.Number, 42);
+
+			assert.deepEqual(
+				field.map((value) => value),
+				[42],
+			);
+		});
+
+		it("string", () => {
+			const field = createPrimitiveField(ValueSchema.String, "Hello world");
+
+			assert.deepEqual(
+				field.map((value) => value),
+				["Hello world"],
+			);
+		});
+
+		// TODO: current types don't allow fluid handle
+		// it("Fluid Handle", () => {
+		// 	const field = createPrimitiveField(ValueSchema.FluidHandle, mockFluidHandle);
```

Comment:
Creating this field fails schema validation. It isn't clear to me how to satisfy the constraints. Guidance would be appreciated 🙂

Reply:
It should just work like the others. `SchemaAware.TypedNode<YourSchema, SchemaAware.ApiMode.Flexible>` might have some bugs, and/or require you to explicitly specify the node ({typenameSymbol: schema.name, valueSymbol: handle}) and type since its old and is from before we restricted node value objects to handles so it couldn't tell nodes and values that could be objects apart.

Redoing the new content insertion APIs is something planned, but for now we are just reusing the old ones.

Reply:
I poked for a bit to no avail. I'm going to remove these for now and leave a TODO to add handle tests in the future.